test/test_model.py

```diff
@@ -816,6 +816,24 @@ def test_stop(self):
             ('stop', ('foo', 'bar')),
         ])
 
+    def test_type_errors(self):
+        meta = ops.charm.CharmMeta.from_yaml("""
+name: k8s-charm
+containers:
+  c1:
+    k: v
+""")
+        # Only the real pebble Client checks types, so use actual backend class
+        backend = ops.model._ModelBackend('myapp/0')
```

Comment:
I'm not sure where this sits, but there are explicit _ModelBackend (a suite focused on _ModelBackend behavior explicitly). it might make more sense to put it there.

Maybe not if the intent is to cover the container.start()/stop methods.

Reply:
Yeah, the intent here is only to cover the type checking in the container start/stop methods.